src/tests/Common/Coreclr.TestWrapper/CoreclrTestWrapperLib.cs

```diff
@@ -209,23 +242,34 @@ public class CoreclrTestWrapperLib
 
         static bool CollectCrashDump(Process process, string crashDumpPath, StreamWriter outputWriter)
         {
-            string coreRoot = Environment.GetEnvironmentVariable("CORE_ROOT");
-            string createdumpPath = Path.Combine(coreRoot, "createdump");
-            string arguments = $"--name \"{crashDumpPath}\" {process.Id} --withheap";
-            Process createdump = new Process();
-            bool crashReportPresent = false;
-
             if (OperatingSystem.IsWindows())
             {
-                createdump.StartInfo.FileName = createdumpPath + ".exe";
-                createdump.StartInfo.Arguments = arguments;
+                return CollectCrashDumpWithMiniDumpWriteDump(process, crashDumpPath, outputWriter);
             }
-            else if (OperatingSystem.IsLinux() || OperatingSystem.IsMacOS())
+            else
             {
-                createdump.StartInfo.FileName = "sudo";
-                createdump.StartInfo.Arguments = $"{createdumpPath} --crashreport {arguments}";
-                crashReportPresent = true;
+                return CollectCrashDumpWithCreateDump(process, crashDumpPath, outputWriter);
             }
+        }
+
+        static bool CollectCrashDumpWithMiniDumpWriteDump(Process process, string crashDumpPath, StreamWriter outputWriter)
+        {
+            using (var crashDump = File.OpenWrite(crashDumpPath))
+            {
+                var flags = DbgHelp.MiniDumpType.MiniDumpWithFullMemory | DbgHelp.MiniDumpType.MiniDumpIgnoreInaccessibleMemory;
+                return DbgHelp.MiniDumpWriteDump(process.Handle, process.Id, crashDump.SafeFileHandle, flags, IntPtr.Zero, IntPtr.Zero, IntPtr.Zero);
+            }
+        }
+
+        static bool CollectCrashDumpWithCreateDump(Process process, string crashDumpPath, StreamWriter outputWriter)
+        {
+            string coreRoot = Environment.GetEnvironmentVariable("CORE_ROOT");
+            string createdumpPath = Path.Combine(coreRoot, "createdump");
+            string arguments = $"--crashreport --name \"{crashDumpPath}\" {process.Id} --withheap";
+            Process createdump = new Process();
+
+            createdump.StartInfo.FileName = "sudo";
```

Comment:
This won't work on windows, right?

Reply:
It won't, but we no longer call this method on Windows. We now call the `CollectCrashDumpWithMiniDumpWriteDump` method on Windows. This method is only called on non-Windows platforms where we can pass the PID to `createdump`.

Reply:
Ah, I see now. How are we planning to dump the stacktrace? Will that happen in a separate PR?

Reply:
Yes, in a later PR after cdb is automatically deployed to the Windows machines.